subsys/bluetooth/host/scan.c

```diff
@@ -2171,3 +2217,8 @@ int bt_le_per_adv_list_clear(void)
 	return 0;
 }
 #endif /* defined(CONFIG_BT_PER_ADV_SYNC) */
+
+bool bt_le_explicit_scanner_running(void)
+{
+	return atomic_test_bit(m_scanner_state.scan_flags, SCAN_ENABLED_REASON_EXPLICIT_SCAN);
+}
```

Comment:
This is a very specific check. 

Would it make sense to make more generic and have something like
```suggestion
bool bt_le_scan_is_reason(enum scan_enabled_reason reason)
{
	return atomic_test_bit(m_scanner_state.scan_flags, reason);
}
```

Reply:
we could do that. However, I don't see a reason why another module would need to check if the scanner is running for a specific reason. The explicit scanner is an exception, because it is not allowed to start the initiator when the application already started the scanner explicitly.
I would propose to leave this as is and update to something more generic once/if that is actually needed.